commit fd0a7832317830a3c2300dbec7953ae187e4a806 (refs/changes/70/1070/1)
Author: Suk Won Choi <sw00.choi@samsung.com>
Date:   2016-12-06T14:42:59+09:00 (2 years, 10 months ago)
    
    joyent/node-docker-registry-client#13 - Add support for no_proxy

diff --git a/lib/registry-client-v1.js b/lib/registry-client-v1.js
index a6c0b7c..253cd22 100644
--- a/lib/registry-client-v1.js
+++ b/lib/registry-client-v1.js
@@ -144,6 +144,7 @@ function login(opts, cb) {
         userAgent: opts.userAgent || common.DEFAULT_USERAGENT,
         agent: opts.agent,
         proxy: opts.proxy,
+        noProxy: opts.noProxy,
         headers: opts.headers,
         rejectUnauthorized: !opts.insecure
     });
@@ -330,6 +331,7 @@ function RegistryClientV1(opts) {
         log: this.log,
         agent: opts.agent,
         proxy: opts.proxy,
+        noProxy: opts.noProxy,
         rejectUnauthorized: !this.insecure,
         userAgent: opts.userAgent || common.DEFAULT_USERAGENT
     };
diff --git a/lib/registry-client-v2.js b/lib/registry-client-v2.js
index fc22c1a..1f9d4d4 100644
--- a/lib/registry-client-v2.js
+++ b/lib/registry-client-v2.js
@@ -202,6 +202,7 @@ function _getToken(opts, cb) {
         log: log,
         agent: opts.agent,
         proxy: opts.proxy,
+        noProxy: opts.noProxy,
         rejectUnauthorized: !opts.insecure,
         userAgent: opts.userAgent || common.DEFAULT_USERAGENT
     });
@@ -673,7 +674,8 @@ function ping(opts, cb) {
         userAgent: opts.userAgent || common.DEFAULT_USERAGENT,
         rejectUnauthorized: rejectUnauthorized,
         agent: opts.agent,
-        proxy: opts.proxy
+        proxy: opts.proxy,
+        noProxy: opts.noProxy
     });
 
     var headers = {};
@@ -875,6 +877,7 @@ function login(opts, cb) {
                 log: log,
                 agent: opts.agent,
                 proxy: opts.proxy,
+                noProxy: opts.noProxy,
                 userAgent: opts.userAgent,
                 insecure: opts.insecure
             }, function (err, token) {
@@ -1037,6 +1040,7 @@ function RegistryClientV2(opts) {
         log: this.log,
         agent: opts.agent,
         proxy: opts.proxy,
+        noProxy: opts.noProxy,
         rejectUnauthorized: !this.insecure,
         userAgent: opts.userAgent || common.DEFAULT_USERAGENT
     };
